.expeditor/scripts/release_habitat/package_and_upload_binary.sh

```diff
@@ -0,0 +1,125 @@
+#!/bin/bash
+
+# Unpack the hart file from our channel, repack it, and upload it to
+# package-router
+
+set -euo pipefail
+
+source .expeditor/scripts/release_habitat/shared.sh
+
+export HAB_AUTH_TOKEN="${ACCEPTANCE_HAB_AUTH_TOKEN}"
+export HAB_BLDR_URL="${ACCEPTANCE_HAB_BLDR_URL}"
+
+channel=$(get_release_channel)
+
+echo "--- Channel: $channel - bldr url: $HAB_BLDR_URL"
+
+release_version=$(get_latest_pkg_release_version_in_release_channel "hab")
+
+declare -g hab_binary
+install_release_channel_hab_binary "x86_64-linux"
+
+${hab_binary} pkg install core/gzip
+${hab_binary} pkg install core/tar
+${hab_binary} pkg install core/wget
+${hab_binary} pkg install core/zip
+
+# Import packages@chef.io GPG signing key
+aws s3 cp s3://chef-cd-citadel/packages_at_chef.io.pgp packages_at_chef.io.pgp --profile=chef-cd
+gpg --import packages_at_chef.io.pgp
+
+#TODO make this nicer
+tmp_root="$(mktemp -d -t "grant-XXXX")"
+extract_dir="$tmp_root/extract"
+mkdir -p "$extract_dir"
+
+echo "--- Downloading $release_version for $BUILD_PKG_TARGET"
+if [[ $BUILD_PKG_TARGET == *"darwin" ]]; then
+  artifact_name=$(buildkite-agent meta-data get MACOS_ARTIFACT)
+  buildkite-agent artifact download "$artifact_name" "$tmp_root/" --step "[:macos: build hab]"
+  mv "$tmp_root/$artifact_name" "$tmp_root/hab-$channel.hart"
+else
+  hab pkg exec core/wget wget "${HAB_BLDR_URL}/v1/depot/pkgs/core/hab/$release_version/download?target=$BUILD_PKG_TARGET" -O "$tmp_root/hab-$channel.hart"
+fi
+
+target_hart="$tmp_root/hab-$channel.hart"
+tail -n+6 "${target_hart}" | \
+    tar --directory "${extract_dir}" \
+        --extract \
+        --xz \
+        --strip-components=6
+
+extracted_hab_binary="$(find "$extract_dir" \( -name hab -or -name hab.exe \) -type f)"
+pkg_target="$(tr --delete '\r' < "${extract_dir}"/TARGET)"
+pkg_arch="$(echo "$pkg_target" | cut -d '-' -f 1)"
+pkg_kernel="$(echo "$pkg_target" | cut -d '-' -f 2)"
+pkg_ident="$(tr --delete '\r' < "$extract_dir"/IDENT)"
+pkg_origin="$(echo "$pkg_ident" | cut -d '/' -f 1)"
+pkg_name="$(echo "$pkg_ident" | cut -d '/' -f 2)"
+pkg_version="$(echo "$pkg_ident" | cut -d '/' -f 3)"
+pkg_release="$(echo "$pkg_ident" | cut -d '/' -f 4)"
+archive_name="hab-$(echo "$pkg_ident" | cut -d '/' -f 3-4 | tr '/' '-')-$pkg_target"
+
+build_dir="$tmp_root/build"
+pkg_dir="$build_dir/${archive_name}"
+
+echo "Copying $extracted_hab_binary to $(basename "$pkg_dir")"
+mkdir -p "$pkg_dir"
+mkdir -p "$tmp_root/results"
+
+if [[ $pkg_target == *"windows" ]]; then
+for file in "$(dirname "$extracted_hab_binary")"/*; do 
+  cp -p "$file" "$pkg_dir/"
+done
+else
+  cp -p "$extracted_hab_binary" "$pkg_dir/$(basename "$extracted_hab_binary")"
+fi
+
+echo "Compressing \`hab' binary"
```

Comment:
Looks like mixed quotes and/or smart quotes here?

Reply:
Yep! copy pasted from bintray uploader :P